libsolidity/ast/Types.cpp

```diff
@@ -2074,7 +2074,13 @@ std::unique_ptr<ReferenceType> ArrayType::copyForLocation(DataLocation _location
 
 BoolResult ArraySliceType::isImplicitlyConvertibleTo(Type const& _other) const
 {
-	if (m_arrayType.location() == DataLocation::CallData && m_arrayType.isDynamicallySized() && m_arrayType == _other)
+	if (
+		auto const* otherArrayType = dynamic_cast<ArrayType const*>(&_other);
+		otherArrayType &&
+		m_arrayType.location() == DataLocation::CallData &&
+		m_arrayType.isDynamicallySized() &&
+		*otherArrayType == *m_arrayType.copyForLocation(otherArrayType->location(), false)
```

Comment:
Can this just be ``m_arrayType->isImplicitlyConvertibleTo(_other)``?
I mean, we basically implement it by calling the array conversion functions, so all cases in which that is fine, should be fine here, shouldn't they?

Reply:
It should work for all cases implemented in array conversion function, yes. Also, coverage there will be full after closing currently open PRs.
I updated to what you suggested, looks much better :-)